pwnlib/tubes/process.py

```diff
@@ -288,11 +294,14 @@ def __init__(self, argv = None,
         #: Environment passed on envp
         self.env = os.environ if env is None else env_val
 
-        if self.executable is None:
-            if shell:
-                self.executable = '/bin/sh'
-            else:
-                self.executable = which(self.argv[0], path=self.env.get('PATH'))
+        if not context.os == "windows":
+            if self.executable is None:
+                if shell:
+                    self.executable = '/bin/sh'
+                else:
+                     self.executable = which(self.argv[0], path=self.env.get('PATH'))
+        else:
+            pass#TODO implement the which for windows
```

Comment:
it should just be `os.environ.get('ComSpec', 'cmd')` instead of `'/bin/sh'`, and the argv should be exactly the same.